docusaurus/docs/cms/plugins-development/widgets-api.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Widgets API
+description: Learn how to use the Widgets API to create and manage widgets in the Strapi admin panel.
+displayed_sidebar: cmsSidebar
+tags:
+  - widgets
+  - admin panel
+  - plugins development
+---
+
+# Widgets API
+
+The Widgets API allows you to create and manage widgets in the Strapi admin panel. Widgets are customizable components that can be added to various parts of the admin interface to display information or provide additional functionality.
+
+## Creating a Widget
+
+To create a widget, you need to use the `Widgets` class from the `@strapi/admin` package. Here's an example of how to create a widget:
+
+```javascript
+import { Widgets } from '@strapi/admin';
+
+const widgets = new Widgets();
+
+widgets.register({
+  id: 'my-widget',
+  pluginId: 'my-plugin',
+  icon: MyWidgetIcon,
+  title: {
+    id: 'my-widget.title',
+    defaultMessage: 'My Widget',
+  },
+  component: () => import('./components/MyWidget'),
```

Comment:
the component function should return a component directly, not a file that has a component as a default export. so:
```suggestion
  component: async () => {
    const { Component } = await import('./components/MyWidget');
    return Component
```